orderer/consensus/follower/follower_chain.go

```diff
@@ -51,16 +143,128 @@ func (*Chain) Errored() <-chan struct{} {
 }
 
 func (c *Chain) Start() {
-	//TODO skeleton
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started || c.stopped {
+		c.logger.Debugf("Not starting because: started=%v, stopped=%v", c.started, c.stopped)
+		return
+	}
+
+	c.started = true
+
+	go c.run()
+
+	c.logger.Info("Started")
 }
 
 func (c *Chain) Halt() {
-	//TODO skeleton
+	c.halt()
+
+	select {
+	case <-c.doneChan:
+	}
+}
+
+func (c *Chain) halt() {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.stopped {
+		c.logger.Debug("Already stopped")
+		return
+	}
+	c.stopped = true
+	close(c.stopChan)
+	c.logger.Info("Stopped")
 }
 
-// StatusReport returns the ClusterRelation & Status
+// StatusReport returns the ClusterRelation & Status.
 func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	clusterRelation := types.ClusterRelationFollower
+	if c.joinBlock != nil {
+		if err := c.amIInChannel(c.joinBlock); err == nil {
```

Comment:
what if you get a block that joins you into the channel? you are no longer a follower, right? In this case i guess this chain no longer exists anymore so this is not relevant right?

Reply:
The follower.Chain also performs all on-boarding to a channel, even if the join-block indicates membership. So in case an orderer gets a join-block for which `c.amIInChannel(c.joinBlock)==nil`, it will still start a follower.Chain, which will pull blocks until the join-block, and then invoke `CreateChannel` and mutate into a `etcdraft.Chain`. (see next PR).

Reply:
Right, but - until we have fully morphed into a real chain I think we should return that we are a follower, no? Otherwise we are reporting a future state.

Reply:
The fact that it is a transient state is carried by the "status=onboarding", I think. Think about the user - he gave the system a block he knows indicates this orderer is a member... 
I think he expects to see "member, onboarding" -> "member, active"

The think the fact that we internally "re-use" the follower chain to do on-boarding does not interest him. That is how I think of it.